v3/src/components/graph/components/chartdots.tsx

```diff
@@ -26,19 +26,23 @@ export const ChartDots = function ChartDots(props: PlotProps) {
   const primaryAttrRole = dataConfiguration?.primaryRole,
     primaryAxisPlace = primaryAttrRole ? attrRoleToAxisPlace[primaryAttrRole] : undefined,
     primaryIsBottom = primaryAxisPlace === 'bottom',
-    primaryAttrID = primaryAttrRole ? dataConfiguration?.attributeID(primaryAttrRole) : '',
+    primaryAttrIDRef = useRef<string>(''),
+    secondaryAttrIDRef = useRef<string>(''),
```

Comment:
FYI, you don't need `<string>` here as the type can be inferred from the default value:
```suggestion
    primaryAttrIDRef = useRef(''),
    secondaryAttrIDRef = useRef(''),
```